packages/enhanced/src/lib/container/runtime/FederationRuntimePlugin.ts

```diff
@@ -90,25 +101,25 @@ class FederationRuntimePlugin {
       });
     }
 
-    const embedRuntimeLines = embedRuntime
-      ? '//using embedded runtime module, will not inject runtime in entry'
-      : Template.asString([
-          `var prevFederation = ${federationGlobal};`,
-          `${federationGlobal} = {}`,
-          `for(var key in federation){`,
-          Template.indent([`${federationGlobal}[key] = federation[key];`]),
-          '}',
-          `for(var key in prevFederation){`,
-          Template.indent([`${federationGlobal}[key] = prevFederation[key];`]),
-          '}',
-        ]);
+    const embedRuntimeLines = Template.asString([
+      `if(!${federationGlobal}.runtime){`,
+      Template.indent([
+        `var prevFederation = ${federationGlobal};`,
+        `${federationGlobal} = {}`,
+        `for(var key in federation){`,
+        Template.indent([`${federationGlobal}[key] = federation[key];`]),
+        '}',
+        `for(var key in prevFederation){`,
+        Template.indent([`${federationGlobal}[key] = prevFederation[key];`]),
+        '}',
+      ]),
+      '}',
+    ]);
 
     return Template.asString([
-      embedRuntime
-        ? ''
-        : `import federation from '${normalizedBundlerRuntimePath}';`,
+      `import federation from '${normalizedBundlerRuntimePath}';`,
       runtimePluginTemplates,
-      embedRuntimeLines,
+      embedRuntime ? '' : embedRuntimeLines,
```

Comment:
do not add runtime if using embed runtime which will already assign to webpack require